osu.Game.Modes.Taiko/LegacyTaikoReplay.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System.Collections.Generic;
+using System.IO;
+using osu.Framework.Input;
+using osu.Game.Input.Handlers;
+using OpenTK.Input;
+
+namespace osu.Game.Modes.Taiko
+{
+    public class LegacyTaikoReplay : LegacyReplay
+    {
+        protected LegacyTaikoReplay()
+        {
+        }
+
+        public LegacyTaikoReplay(StreamReader reader)
+            : base(reader)
+        {
+        }
+
+        public override ReplayInputHandler CreateInputHandler() => new LegacyTaikoReplayInputHandler(Frames);
+
+        private class LegacyTaikoReplayInputHandler : LegacyReplayInputHandler
+        {
+            public LegacyTaikoReplayInputHandler(List<LegacyReplayFrame> replayContent)
+                : base(replayContent)
+            {
+            }
+
+            public override List<InputState> GetPendingStates() => new List<InputState>
+            {
+                new InputState
+                {
+                    Keyboard = new ReplayKeyboardState(new List<Key>(new[]

```

Comment:
you're doing the array in list thing again.

also i'd prefer you didn't add `Key.Unknown` potentially four times, and just move the list construction above with conditional `Add`s